lib/puppet/functions/katello/build_dn.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require 'yaml'
+# @summary
+#   Convert an array of attribute pairs to a DN string, ignoring empty values
+#
+# @example Pass a set of
+#   $client_dn = katello::build_dn([['CN', 'foo.example.com'], ['O', 'my_org']])
+Puppet::Functions.create_function(:'katello::build_dn') do
+  # @param options
+  #
+  # @return String
```

Comment:
Technically you need to write
```suggestion
  # @return [String] human readable description
```

However, I prefer to use `return_type 'String'` in the dispatch, similar to `param`.